File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    visited = [words[0]];
    pre = words[0][-1];
    
    for i in range(1, len(words)):
        if words[i] in visited: return [i % n + 1, i // n + 1]
    
        next = words[i][0];
        if pre == next: 
            visited.append(words[i]);
            pre = words[i][-1];
        else: return [i % n + 1, i // n + 1]
    
    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    const visited = [words[0]];
    let pre = words[0][words[0].length - 1];
    
    for(let i = 1; i < words.length; i++){
        if(visited.includes(words[i])) return [i % n + 1, Math.floor(i / n) + 1]
        
        let next = words[i][0];
        if(pre === next) {
            visited.push(words[i])
            pre = words[i][words[i].length - 1]
        }
        else return [i % n + 1, Math.floor(i / n) + 1]
    }
    return [0, 0]
}